### PR TITLE
Check slice thickness

### LINKED ISSUE #2: [Feat] check the slice thickness is uniform or variable and trigger different workflows.
- [x] From Dicom header check the slice thickness type
- [x] If uniform thickness go with processing
- [x] else for now raise not implemented error

### PR DESCRIPTION
**Pull request by a-parida12**
Solves Issue & Closes #2 

This feature checks for variable slice thickness and rejects them

### CODE REVIEW

File: nekton/dcm2nii.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import glob
+import os
+from typing import List
+
+import pydicom
+
+from utils.dicom import is_file_a_dicom
+from base import baseConverter
```
Comment (yigitsoy) [inline nekton/dcm2nii.py]: Consistent naming convention, e.g. UpperCamelCase for classes

File: nekton/dcm2nii.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+class Dcm2Nii(baseConverter):
+    def __init__(self):
+        super().__init__()
+
+    def get_all_dicoms(self, dicom_directory: str) -> List[str]:
```
Comment (yigitsoy) [inline nekton/dcm2nii.py]: this could also be a static method I guess

File: nekton/dcm2nii.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        )
+        return False if len(all_slice_thickness) == 1 else True
+
+    def _run_conv_variable(self, all_dcm_paths: List[str], name: str) -> str:
+        raise NotImplementedError(
+            "DICOM with variable slice thickness cannot to be handled yet!!"
+        )
+        return ""
+
+    def _run_conv_uniform(self, all_dcm_paths: List[str], name: str) -> str:
```
Comment (yigitsoy) [inline nekton/dcm2nii.py:70]: similarly here, if there is no need for using self, do not use it, but if you need to keep instance data, then, it makes sense.

File: nekton/dcm2nii.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            all_dcm_paths = self.get_all_dicoms(dicom_directory)
+        except Exception as err:
+            raise RuntimeError(f"Error parsing dicoms: {err}")
+
+        if self.check_slice_thickness_variable(all_dcm_paths):
+            converted_file_path = self._run_conv_variable(all_dcm_paths, name)
+        else:
+            converted_file_path = self._run_conv_uniform(all_dcm_paths, name)
+
+        return converted_file_path
```
Comment (yigitsoy) [inline nekton/dcm2nii.py]: I would add a general try/catch here in case subsequent calls fail

File: nekton/utils/json_helpers.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    Args:
+        dictionary (dict): a dictionary that has to be converted to json
+        path (str): path to output json file
+
+    Returns:
+        str: path to output json file
+    """
+    json_object = json.dumps(dictionary, indent=4)
+    with open(path, "w") as outfile:
+        outfile.write(json_object)
```
Comment (yigitsoy) [inline nekton/utils/json_helpers.py]: to be correct, I would add a try/catch here too

File: nekton/utils/json_helpers.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    Args:
+        path (str): path to a json file
+
+    Returns:
+        dict: the dictionary read from a json file
+    """
+    with open(path, "r") as openfile:
+        json_object = json.load(openfile)
+
+    return json_object
```
Comment (yigitsoy) [inline nekton/utils/json_helpers.py]: the same..

**Review by yigitsoy — CHANGES_REQUESTED** [review]
Please consider the points provided in the comments.

File: nekton/dcm2nii.py (hunk 3 above)
Comment (a-parida12) [reply nekton/dcm2nii.py:70]: This is part of another feature and will not be a static method, but left it for later

**Review by yigitsoy — APPROVED** [review]

**Pull request merged**